### PR TITLE
Feat: add completion toplevel decorators

### PR DESCRIPTION
**Pull request by aspeddro**
image

### CODE REVIEW

File: analysis/tests/src/expected/Completion.res.txt [hunk 1]
```diff
@@ -564,7 +564,7 @@ Completable: Cdecorator(react.)
 Package opens Pervasives.JsxModules.place holder
 Resolved opens 1 pervasives
 [{
-    "label": "component",
+    "label": "react.component",
```
Comment (aspeddro) [inline analysis/tests/src/expected/Completion.res.txt:567]: I don't understand this change, but it didn't affect autocomplete for `@react.component`

https://github.com/rescript-lang/rescript-vscode/assets/16160544/f0fa547e-056d-4219-8371-884242e15519

File: analysis/src/CompletionDecorators.ml [hunk 2]
```diff
@@ -1,6 +1,8 @@
+type kind = TopLevel of string | Local of string | TopLevelOrLocal of string
```
[the remark this answers is not in the document]
Comment (aspeddro) [reply analysis/src/CompletionDecorators.ml]: Done https://github.com/rescript-lang/rescript-vscode/pull/799/commits/4a4a3520ec1bdbe53c6c5413031367736c91d605

File: analysis/src/CompletionDecorators.ml [hunk 3]
```diff
@@ -30,8 +30,6 @@ It is also possible to map a ReScript record to a JavaScript array by passing in
       [
         {|The `@deprecated` decorator is used to add deprecation notes to types, values and submodules. The compiler and editor tooling will yield a warning whenever a deprecated entity is being used.
 
-Alternatively, use the `@@deprecated` decorator to add a deprecation warning to the file level.
```
Comment (zth) [inline analysis/src/CompletionDecorators.ml:33]: Should this really be removed?

**Review by zth — APPROVED** [review]
Great! 👍 Just one small comment.

File: analysis/src/CompletionDecorators.ml (hunk 3 above)
Comment (aspeddro) [reply analysis/src/CompletionDecorators.ml]: Add here https://github.com/rescript-lang/rescript-vscode/pull/799/commits/e3e0463259bc85ee4ac86ebf71ec3fc581ebda33

**Pull request merged**